pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -408,8 +410,30 @@ func (is *InstanceService) InstanceCreate(clusterName string, name string, confi
 		SecurityGroups:   config.SecurityGroups,
 		ServiceClient:    is.computeClient,
 	}
+
+	// If the root volume Size is not 0, means boot from volume
+	if config.RootVolume.Size != 0 {
```

Comment:
Should this check be `if config.RootVolume != nil && config.RootVolume.Size != 0` ?

I believe, as is, this will fail if `rootVolume` is omitted from the `MachineSpec`

Reply:
yes, you are right